packages/byods/src/data-source-client/index.ts

```diff
@@ -86,4 +97,126 @@ export default class DataSourceClient {
   public async delete(id: string): Promise<ApiResponse<void>> {
     return this.httpClient.delete<void>(`${DATASOURCE_ENDPOINT}/${id}`);
   }
+
+  /**
+   * This method refreshes the DataSource token using dataSourceId and tokenLifetimeMinute
+   * @param {string} dataSourceId The id of data source
+   * @param {number} tokenLifetimeMinutes The Life time minutes for the data source. <=1440
+   * @returns {Promise<void>}
+   * @example
+   * const dataSourceClient = new DataSourceClient(httpClient); Create an instance of dataSourceClient by passing instance of HttpClient.
+   * Start auto-refresh
+   * const { promise, cancel } = dataSourceClient.refreshDataSourceToken('12345', 30); Invoke refreshDataSourceToken method by passing dataSourceId and tokenLifetimeMinutes.
+    promise
+    .then(() => {
+        console.log('Auto-refresh started successfully.');
+    })
+    .catch((error) => {
+      console.error('Failed to start auto-refresh:', error);
+    });
+
+    Way to Cancel the auto-refresh if needed
+    cancel();
+   */
+
+  public async refreshDataSourceToken(
+    dataSourceId: string,
+    tokenLifetimeMinutes?: number
+  ): Promise<{promise: Promise<void>; cancel: () => void}> {
```

Comment:
return type doesn't seem to be right? we just need to send an object containing cancel method.

Reply:
Addressed